SCF/SCF-Session8-ZooProblem/Zone.java

```diff
@@ -0,0 +1,79 @@
+import java.util.*;
+
+public class Zone {
+
+	private String animalCategory;
+	private int maxCage;
+	private List<Cage> cages = new LinkedList<Cage>();
+	private boolean hasPark;
+	private boolean hasCanteen;
+	private int numberOfCagesLeft ;
+	
+	Zone(String category, int maxCage, boolean hasPark, boolean hasCanteen){
+		this.animalCategory = category;
+		this.maxCage = maxCage;
+		this.hasPark = hasPark;
+		this.hasCanteen = hasCanteen;
+		this.numberOfCagesLeft = maxCage;
+	}
+	
+	/**
+	 * 
+	 * @param animal - Animal Class object
+	 * Add animal to Zone
+	 */
+	public void addAnimalToZone(Animal animal) {
+		if(this.numberOfCagesLeft == this.maxCage) {
+			Cage cage = new Cage(animal.getCategory(), 1);
+			cage.addAnimalToCage(animal);
+			this.cages.add(cage);
+			this.numberOfCagesLeft --;
+		}
+		else {
+			boolean added = false;
+			for(Cage cage: cages) {
+				if(cage.getCageCategory().equals(animal.getCategory())) {
+					if(cage.isSpaceInCage()) {
+						cage.addAnimalToCage(animal);
+						added = true;
+						break;
+					}
+				}
+			}
+			if(!added) {
+				if(this.numberOfCagesLeft == 0) {
+					throw new AssertionError("Cages Limit full");
+				}
+				else {
+					Cage cage = new Cage(animal.getCategory(), 1);
```

Comment:
have some default maxLimit for cage, otherwise every time you will need to create a new cage for each animal which is not good for management perspective.